### PR TITLE
[AWS SES] Add datadog counters and error tracking to bounced email manager

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAASP-10229

##### SUMMARY
This adds some additional tracking in datadog for any errors that come up during the `process_bounced_emails` periodic task and counts the number of different error types when processing emails.

### CODE REVIEW

File: corehq/util/bounced_email_manager.py [hunk 1]
```diff
@@ -137,6 +138,7 @@ def _get_aws_info(self, message, uid):
                     destination=mail_info.get('destination', []),
                 ))
         else:
+            datadog_counter('bounced_email_manager.unknown_notification_type')
```
Comment (millerdev) [inline corehq/util/bounced_email_manager.py]: How did you evade the prefix enforcer (metric name should start with `commcare.`)? Tests should have failed, but maybe there are no tests that trigger these error conditions?

https://github.com/dimagi/commcare-hq/blob/786db7e000394bcef2cad353d68f0c1521d4dc34/corehq/util/datadog/gauges.py#L120-L125

**Review by millerdev — COMMENTED** [review]
Looks like all of the added datadog counters are for error conditions. What do you think of adding one that counts _each bounced message_? That would make it easy to tell how many errors are occurring relative to the total number of bounced emails being processed.

File: corehq/util/bounced_email_manager.py (hunk 1 above)
Comment (biyeun) [reply corehq/util/bounced_email_manager.py]: updated

File: corehq/util/bounced_email_manager.py [hunk 2]
```diff
@@ -137,6 +138,9 @@ def _get_aws_info(self, message, uid):
                     destination=mail_info.get('destination', []),
                 ))
         else:
+            datadog_counter(
+                'commcare.util.bounced_email_manager.unknown_notification_type'
```
Comment (dannyroberts) [inline corehq/util/bounced_email_manager.py]: nit: I'd omit `util` here (and below) and go with just `commcare.bounced_email_manager.unknown_notification_type`. I say that assuming the `util` is there to mirror the file structure, and that's not really a convention we have, and wouldn't hold up well under a refactor, etc.

**Review by dannyroberts — APPROVED** [review]

File: corehq/util/bounced_email_manager.py (hunk 2 above)
Comment (biyeun) [reply corehq/util/bounced_email_manager.py]: done

**Review by millerdev — COMMENTED** [review]
@biyeun did you miss this question?

> What do you think of adding one that counts each bounced message? That would make it easy to tell how many errors are occurring relative to the total number of bounced emails being processed.

**Comment by biyeun** [comment]
@millerdev yep I missed that sorry. adding that in now

**Review by millerdev — APPROVED** [review]

**Pull request merged**